### DISCUSSION TITLE
No Configuration Methods Succeeded

### DISCUSSION THREAD

**Discussion by rvinf** [ipxe/ipxe#1277] [General]
Am using rev `77b07ea4fdc259d7253c6f9df2beda6e6c7a9d85` to create UEFI iPXE file. During boot the server picks up the efi file but later when it is supposed to get the dynamic boot file from the PXE server, it fails with No Configuration Methods Succeeded as shown in the pic below.

I tried using the latest revision but they fail with complication errors. Any thoughts into this will be greatly helpful.

<img width="1076" alt="Screenshot 2024-08-19 at 20 38 58" src="https://github.com/user-attachments/assets/be1b00f2-d236-4fc8-800d-c74bc528203f">

**Comment by NiKiZe** [comment]
Did you follow the URL in the message and followed the suggestions there?

**Reply by rvinf** [reply]
Yes, the routine troubleshooting steps like, check with generic efi file, check proxy settings. examine tftp options etc.,

Adding to the above: I have 2 types of hardware at my organization (Dell R740 and R650xs) This same efi works fine with R740 but not on R650xs.

**Reply by NiKiZe** [reply]
I understand that you did generic checks, but did you read https://ipxe.org/040ee186 ?

In this case however, I would suggest that you press Ctrl-B to get iPXE shell on both the working and non working machine and post output.

I can also see that net2-30 is used, which I assume is a vlan configuration in firmware?

What I can't fully deduce is if you are using a ipxe.efi, snp.efi, or snponly.efi based build.
Neighter can I be sure if you are using embedded script or not.

**Reply by rvinf** [reply]
- Per the link , I tried with more the last 10 revs. 
- Am using ipxe.efi
- Yes am using embedded chain script. 
However, one thing I didn't try is getting into iPXE shell, for some reason am unable to get there by press Ctrl-B or any other key combinations I could find.

I'll now try using the  `DEBUG=ifmgmt` option. 

Thanks in advance for all the assistance @NiKiZe .

**Reply by rvinf** [reply]
recomplied using `make -j4 bin-x86_64-efi/ipxe.efi EMBED=../script.ipxe  DEBUG=ifmgmt ` and tried, but nothing changed, neither do I see any debug information anywhere.

**Reply by NiKiZe** [reply]
Then you need to post the script you are using.
If you have embedded script you will need to handle the Ctrl-B on your own, but do add `ifstat` to the script so we can get the requested output from both working and non working.

And especially, do verify that the NIC you are using is connected to the correct network, and that the vlan is configured properly.

The next step would otherwise be to pull out wireshark and verify what the machine sends out the wire. (Or check the logs on the DHCP server for the discover packets)

**Reply by rvinf** [reply]
@NiKiZe Got it resolved, compiling with:

`make -j4 **NO_WERROR=1** bin-x86_64-efi/ipxe.efi EMBED=../script.ipxe`  on rev: `eecb75ba4809bd5d3e6d63413a45c5fccbda2bc2`  fixed it.

Thanks for all the assistance.

**Reply by NiKiZe** [reply]
So you are saying that building a new commit worked. And your actual issue was that you had build issues, but didn't post the build log, or version of GCC that you are using?

**Reply by rvinf** [reply]
My apologies for the delay in replying. Build log didn't show any issues, I didn't change the GCC version hence I didn't look at log file.
